### PR TITLE
Implement Training Log creation

### LINKED ISSUE #9: [Backend] - Add functionality to Create a Training Log
### Description
Each service animal is able to have multiple training logs associated. Each training log indicates details regarding a training session for a certain training animal. We need functionality to add a training log to our database and have it associated with the respective animal.

### Acceptance Criteria

**Mongoose & Typescript Types**
- [x]  Create the mongoose schema for a TrainingLog + adjust the typescript types as needed

**Backend Action**
- [x] Create a backend action (MongoDB query) that will add the training log to the TrainingLog collection in Mongo (`backend/server/mongodb/actions/TrainingLog.ts`)

**API Endpoint**
- [x] Create a POST request endpoint at `/api/training.ts` and use the backend action to create the training log based on a given input object of type TrainingLog (`backend/src/pages/api/training.ts`)
- [x] Note: this endpoint will require authentication so make sure to set those parameters to true in the config dictionary; the user will also need to have the role of `Role.NONPROFIT_USER`

**Frontend Action**
- [x] Create a frontend action that calls this API endpoint (`mobile/actions/TrainingLog.ts`)
- [x] **Important: Use the internal web request and API wrappers -- this greatly simplifies middleware + authentication for you!**

### Other Notes
- Any types you need can be found in `mobile/utils/types.ts` or `backend/src/utils/types.ts`
- The backend for this repository has a bit of a learning curve but it will speed up development in the long run; I highly recommend you read through some of the other files but the general flow of data is: Backend Action (MongoDB Operations) --> API Wrapper --> Frontend Action --> User Interface
- If you need to get user information, you can use the getUser method in backend/server/utils/Authentication/ to get the user information. It requires an access token which should be in the request header (i.e. req.headers.accesstoken). Using this will return you a user object if the user exists in the database
- You can assume we will have all the required fields and some or all of the optional fields for the training log for this endpoint
- Please bother me if you have any questions 😃

### PR DESCRIPTION
**Pull request by jsn-li**
## What is this?
Implement schema, actions for creating Training Logs.

## Setup
- [x] No special setup needed

## Steps to Test
- [x] Start the backend
- [x] Make a post request to `http://localhost:3000/api/training` with a body following this example: 
```json
{
    "date": "2022-09-11",
    "description": "description",
    "skills": [
        "skill1",
        "skill2"
    ],
    "trainingHours": 5,
    "behavior": {
        "description": "behaviorDescription",
        "repeat": 3
    },
    "animal": "animal",
    "video": "videoUrl"
}
```

## Associated Issues
Closes #9 

## Additional Instructions
Since some fields were dependent on the Service Animal schema issue #6, those types need to be changed to the actual types implemented in that ticket. These fields are marked with TODOs.

Additionally, while I set all of the `video` fields to required, I did not implement any defaults (i.e. defaulting to ""). I think this should be implemented in the actual frontend logic. I think I also am leaning towards keeping it optional with no default, and having the field ultimately being omitted from the DB entry. Any feedback would be appreciated here :)

### CODE REVIEW

File: backend/server/mongodb/models/TrainingLog.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    type: Number,
+    required: true,
+  },
+  behavior: {
+    type: ServiceAnimalBehaviorSchema,
+    required: true,
+  },
+  animal: {
+    // TODO: replace with the actual schema
+    type: Object,
```
Comment (SamratSahoo) [inline backend/server/mongodb/models/TrainingLog.ts]: I am thinking we can just change this to an ObjectId (as opposed to a ServiceAnimal Object) -- I don't see us needing to access a ServiceAnimal object directly from the training log so a reference would probably be fine

You will need to change the Type in backend/src/utils/types.ts and mobile/utils/types.ts and the places where this model is referenced.

File: backend/server/mongodb/models/TrainingLog.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    required: true,
+  },
+  video: {
```
Comment (SamratSahoo) [inline backend/server/mongodb/models/TrainingLog.ts:36]: I think I can agree with keeping this optional and omitted -- tbh it doesn't really matter but if we want to keep things organized with our DB, then omitting it seems smart.

File: backend/server/mongodb/models/TrainingLog.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    type: String,
+    required: true,
+  },
+});
```
Comment (SamratSahoo) [inline backend/server/mongodb/models/TrainingLog.ts]: I think I can agree with keeping this optional and omitted -- tbh it doesn't really matter but if we want to keep things organized with our DB, then omitting it seems smart.

**Comment by jsn-li** [comment]
I added a commit to change the mobile eslint version to match the backend. This resolves some issues I was having with the mobile eslint not working. If this causes issues I can revert

**Review by SamratSahoo — APPROVED** [review]
Looks good

**Pull request merged**